tests/Resources/ti.utils.test.js

```diff
@@ -0,0 +1,135 @@
+/*
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2015-Present by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+/* eslint-env mocha */
+/* global Ti */
+/* eslint no-unused-expressions: "off" */
+'use strict';
+var should = require('./utilities/assertions');
+
+describe('Titanium.Utils', function () {
+	it('exists', function () {
+		should(Ti.Utils).not.be.undefined;
+		should(Ti.Utils).be.an.Object;
+	});
+
+	it('apiName', function () {
+		should(Ti.Utils).have.readOnlyProperty('apiName').which.is.a.String;
+		should(Ti.Utils.apiName).be.eql('Ti.Utils');
+	});
+
+	it('#base64decode(String)', function () {
+		// Basic tests
+		should(Ti.Utils.base64decode).be.a.Function;
+		var test = Ti.Utils.base64decode('dGVzdA==');
+		should(test).be.a.Object;
+		should(test.apiName).eql('Ti.Blob');
+		should(test.getText()).be.eql('test');
+
+		// Test string without padding
+		should(!Ti.Utils.base64decode('eyJzdWIiOiJ0ZXN0IiwiZW1haWwiOiJ0ZXN0IiwiYXVkIjoidGVzdCIsImp0aSI6ImxvTHM4d2o5aWxBQUtWckNxbzhaMFMiLCJpc3MiOiJodHRwczpcL1wvc3NvLmV4YW1wbGUuY29tIiwiaWF0IjoxNTI2MTY3NDc3LCJleHAiOjE1MjYxNjc0NzcsInBpLnRlc3QiOiJMSTRmMW81Q2pqU2tHU2xTanM0bHlPeVlROCJ9')).not.be.null;
+
+	  // More padding tests 
+	  should(Ti.Utils.base64decode('Zg').text).eql('f');
+	  should(Ti.Utils.base64decode('Zm8').text).eql('fo');
+	  should(Ti.Utils.base64decode('Zm9v').text).eql('foo');
+	  should(Ti.Utils.base64decode('Zm9vYg').text).eql('foob');
+	  should(Ti.Utils.base64decode('Zm9vYmE').text).eql('fooba');
+	  should(Ti.Utils.base64decode('Zm9vYmFy').text).eql('foobar');
+	});
+
+	// FIXME Windows gives: 'base64decode: attempt to decode a value not in base64 char set'
+	it.windowsBroken('#base64decode(Ti.Blob)', function () {
+		var f = Titanium.Filesystem.getFile(Titanium.Filesystem.resourcesDirectory, 'txtFiles/encodedFile.txt'),
+			blob = Ti.Utils.base64decode(f.read());
+		should(blob.toString()).eql('Decoding successful!');
+	});
+
+	it('#base64encode(String)', function () {
+		var test;
+		should(Ti.Utils.base64encode).be.a.Function;
+		test = Ti.Utils.base64encode('test');
+		should(test).be.a.Object;
+		should(test.apiName).eql('Ti.Blob');
+		should(test.getText()).be.eql('dGVzdA==');
+	});
+
+	it('#base64encode(Ti.Blob)', function () {
```

Comment:
Add test for `#base64encode(Ti.Filesystem.File)`

Reply:
Already in https://github.com/appcelerator/titanium_mobile/pull/9773/files#diff-6384f93cac05d6aea1061d799b0fe88aR67. Other test is added.